AdminUi/src/AdminUi/ClientApp/src/app/components/quotas/assign-quotas-dialog/assign-quotas-dialog.component.ts

```diff
@@ -62,6 +64,56 @@ export class AssignQuotasDialogComponent {
         });
     }
 
+    public createQuota(quotaData: AssignQuotaData): void {
+        this.loading = true;
+
+        if (this.data.address) {
+            this.createIdentityQuota(this.data.address, quotaData);
+        } else if (this.data.tierId) {
+            this.createTierQuota(this.data.tierId, quotaData);
+        }
```

Comment:
This is an ugly hack. `data` is (as far as I can see) of type `any`. There has to be a different way.